src/main/java/com/dreammaster/modfixes/thaumcraft/ReuseWardFocusFix.java

```diff
@@ -0,0 +1,57 @@
+package com.dreammaster.modfixes.thaumcraft;
+
+import com.dreammaster.modfixes.ModFixBase;
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+import cpw.mods.fml.common.gameevent.PlayerEvent;
+import cpw.mods.fml.common.registry.GameRegistry;
+import net.minecraft.block.Block;
+import net.minecraft.item.ItemStack;
+import thaumcraft.common.tiles.TileArcaneWorkbench;
+
+public class ReuseWardFocusFix extends ModFixBase {
+    private ItemStack wardedGlass, focusWard;
+
+    public ReuseWardFocusFix() {
+        super("ReuseWardFocusFix");
+    }
+
+    @Override
+    public boolean needsForgeEventBus() {
+        return false;
+    }
+
+    @Override
+    public boolean needsFMLEventBus() {
+        return true;
+    }
+
+    @Override
+    public boolean getIsActive() {
+        return wardedGlass != null && focusWard != null;
+    }
+
+    @Override
+    public boolean init() {
+        Block wardedGlassBlock = GameRegistry.findBlock("Thaumcraft", "blockCosmeticOpaque");
+        if (wardedGlassBlock == null)
+            return false;
+        wardedGlass = new ItemStack(wardedGlassBlock, 1, 2);
+        return (focusWard = GameRegistry.findItemStack("Thaumcraft", "FocusWarding", 2)) != null;
+    }
+
+    @SubscribeEvent
+    public void onItemCrafted(PlayerEvent.ItemCraftedEvent e) {
+        if (getIsActive()) {
+            if (e.craftMatrix instanceof TileArcaneWorkbench && wardedGlass.isItemEqual(e.crafting)) {
```

Comment:
That is not enough for arcane crafting terminal and arcane assembler, perhaps just check if the crafting inventory is not null?